src/Apps/Order/Routes/Shipping2/Hooks/useShippingContextHelpers.tsx

```diff
@@ -0,0 +1,185 @@
+import { useCallback, useContext, useState } from "react"
+import { FormikProps } from "formik"
+import { ParsedOrderData } from "Apps/Order/Routes/Shipping2/Hooks/useParseOrderData"
+import {
+  ShippingContext,
+  ShippingStage,
+} from "Apps/Order/Routes/Shipping2/Utils/ShippingContext"
+import { FulfillmentValues } from "Apps/Order/Routes/Shipping2/Utils/shippingUtils"
+import createLogger from "Utils/logger"
+import { useOrderTracking } from "Apps/Order/Utils/useOrderTracking"
+import {
+  ErrorDialogs,
+  getErrorDialogCopy,
+} from "Apps/Order/Utils/getErrorDialogCopy"
+import { Dialog } from "Apps/Order/Dialogs"
+import { ArtaErrorDialogMessage } from "Apps/Order/Routes/Shipping2/ArtaErrorDialogMessage"
+
+export const useShippingContext = () => {
+  return useContext(ShippingContext)
+}
+
+type FulfillmentHelpers = Pick<
+  FormikProps<FulfillmentValues>,
+  "submitForm" | "isValid" | "values"
+>
+
+export interface ShippingContextHelpers {
+  dialog: Dialog
+  orderTracking: ReturnType<typeof useOrderTracking>
+  fulfillmentDetails: Pick<
+    FormikProps<FulfillmentValues>,
+    "submitForm" | "isValid" | "values"
+  > & {
+    setFulfillmentFormHelpers: (helpers: FulfillmentHelpers) => void
+  }
+  handleExchangeError: (
+    error: {
+      code: string
+      data: string | null | undefined
+    },
+    logger: ReturnType<typeof createLogger>
+  ) => void
+  setSelectedShippingQuote: (payload: string | null) => void
+  setNewSavedAddressId: (payload: string | null) => void
+  setStage: (payload: ShippingStage) => void
+}
+
+export const useShippingContextHelpers = (
+  parsedOrderData: ParsedOrderData,
+  dialog: Dialog,
+  dispatch: (action: any) => void
+): ShippingContextHelpers => {
+  /**
+   * Because there is a single button for both fulfillment details and
+   * shipping quote steps (and duplicated in the sidebar)
+   * we need to hack some formik values UP from the fulfillment details form.
+   *
+   * Currently we need to pass up:
+   */
+  const [fulfillmentFormHelpers, setFulfillmentFormHelpers] = useState<
+    Pick<FormikProps<FulfillmentValues>, "submitForm" | "isValid" | "values">
+  >({
+    // Used to submit the form
+    submitForm: () => Promise.reject(new Error("form not loaded")),
+    // Used to disable the button
+    isValid: false,
+    // Used to get the form values for un-saving the address if the user
+    // unchecks it after saving it in the fulfillment details step.
+    values: ({
+      attributes: {
+        saveAddress: false,
+      },
+    } as unknown) as FulfillmentValues,
+  })
+  const fulfillmentDetails = {
+    ...fulfillmentFormHelpers,
+    setFulfillmentFormHelpers,
+  }
+
+  const orderTracking = useOrderTracking()
+  const isArtsyShipping = !!parsedOrderData.savedFulfillmentDetails
+    ?.isArtsyShipping
+
+  const handleExchangeError = useCallback(
+    (
+      error: { code: string; data: string | null | undefined },
+      logger: ReturnType<typeof createLogger>
+    ) => {
```

Comment:
This was [handleSubmitError](https://github.com/artsy/force/pull/13246/files#diff-f6595a96ecf910d28cf63a298ec8347b3755f89334c02b4fa3b618f916ce10b4L152) in the index route. It is only used for exchange error codes (both shipping quotes and fulfillment details).